### PR TITLE
Exception Handling in target_set_cancel and grpc timeout 

### PR DESCRIPTION
**Pull request by henry54809**

### CODE REVIEW

File: daq/runner.py [hunk 1]
```diff
@@ -211,7 +211,10 @@ def _init_daq_run_id(self):
     def _init_device_result_client(self):
         server_port = self.config.get('device_reporting', {}).get('server_port')
         if server_port:
-            return DeviceReportClient(server_port=server_port)
+            timeout = self.config['device_reporting'].get('rpc_timeout_sec')
+            if timeout:
+                return DeviceReportClient(server_port=server_port, rpc_timeout_sec=timeout)
```
Comment (grafnu) [inline daq/runner.py:216]: NB: I'm surprised Python doesn't have a more elegant way of doing this!  I've run into the same problem before...

File: proto/system_config.proto [hunk 2]
```diff
@@ -253,4 +253,5 @@ message Interface {
 message DeviceReporting {
   // Server port
   int32 server_port = 1;
+  int32 rpc_timeout_sec = 2;
```
Comment (grafnu) [inline proto/system_config.proto:256]: add comment

File: daq/device_report_client.py [hunk 3]
```diff
@@ -28,15 +31,28 @@ def send_device_result(self, mac, device_result):
...
         }
-        for port_event in self._stub.GetPortState(dict_proto(device, Device)):
+        result_generator = self._stub.GetPortState(dict_proto(device, Device))
+        def cancel_request():
+            result_generator.cancel()
+
+        self._pending_requests.append(cancel_request)
+        for port_event in result_generator:
             callback(port_event)
+        if cancel_request in self._pending_requests:
```
[the remark this answers is not in the document]
Comment (henry54809) [reply daq/device_report_client.py]: Originally I had a different place where handlers would be removed.

**Comment by henry54809** [comment]
PTAL.

File: proto/system_config.proto [hunk 4]
```diff
@@ -253,5 +253,6 @@ message Interface {
 message DeviceReporting {
   // Server port
   int32 server_port = 1;
+  // GRPC timeout in seconds
```
Comment (grafnu) [inline proto/system_config.proto:256]: nit: blank line before

**Review by grafnu — APPROVED** [review]

**Pull request merged**